Change some log to debug

YuanTingHsieh opened the pull request:
Solve #238 

- Change some logs' logging level to DEBUG to decrease default log file size
- Remove unused / duplicate codes

ZiyueXu77 commented on nvflare/private/fed/server/fed_server.py:
```diff
@@ -379,9 +379,7 @@ def GetTask(self, request, context):
-                self.logger.info(f"Return task:{taskname} to client:{client.name} --- ({token}) ")
-
-                # self.fl_ctx.merge_sticky(fl_ctx)
+                self.logger.info(f"GetTask: Return task: {taskname} to client: {client.name} ({token}) ")
```
I think we can add an if here: if the taskname is "__try_again__", then make the logger to debug, otherwise stay info. Similar handling can be applied to other cases

YuanTingHsieh replied:
So you suggest we move all logging related to the concept of "try_again" into debug level.

On one hand I feel that adding an "if" statement for the "try_again" is not that clean.

What are your thoughts @holgerroth @yanchengnv ?

holgerroth replied:
debug log level makes sense to me.

ZiyueXu77 commented on nvflare/private/fed/server/server_runner.py:
```diff
@@ -201,77 +201,45 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> (str, str,
 
                 if not task_name or task_name == SpecialTaskName.TRY_AGAIN:
-                    self.log_info(fl_ctx, "no task currently for client - asked client to try again later")
+                    self.log_error(fl_ctx, "no task currently for client - asked client to try again later")
```
this will always show under "try_again", I think we can move it to debug

holgerroth replied:
Yes, debug log level makes sense to me.

holgerroth commented on nvflare/private/fed/server/server_runner.py:
```diff
@@ -201,77 +201,45 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> (str, str,
 
         if self.status == "init":
-            self.log_info(fl_ctx, "server runner still initializing - asked client to try again later")
+            self.log_error(fl_ctx, "server runner still initializing - asked client to try again later")
```
shouldn't this be debug level as well?

holgerroth commented on nvflare/private/fed/server/server_runner.py:
```diff
@@ -201,77 +201,45 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> (str, str,
         peer_ctx = fl_ctx.get_peer_context()
         if not isinstance(peer_ctx, FLContext):
-            self.log_error(fl_ctx, "invalid task request: no peer context, asked client to try again.")
+            self.log_error(fl_ctx, "invalid task request: no peer context - asked client to try again later")
```
should this be error or debug?

YuanTingHsieh replied:
changed to debug as suggested.

holgerroth commented on nvflare/private/fed/server/server_runner.py:
```diff
@@ -201,77 +201,45 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> (str, str,
             with self.wf_lock:
                 if self.current_wf is None:
-                    self.log_info(fl_ctx, "There's no current workflow - asked client to try again later")
+                    self.log_error(fl_ctx, "no current workflow - asked client to try again later")
```
Also debug?

YuanTingHsieh replied:
changed back to info as Yan suggested.

holgerroth reviewed (changes requested):
Looks good but I left some comments.

ZiyueXu77 commented:
Also, the "check status from process listener......" output from [nvflare/private/fed/client/client_executor.py](https://github.com/NVIDIA/NVFlare/blob/da80cb206af751392e34bf6de36eef4818d4cfd7/nvflare/private/fed/client/client_executor.py)

yanchengnv commented on nvflare/private/fed/server/server_runner.py:
```diff
@@ -201,77 +201,45 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> (str, str,
-                            fl_ctx,
-                            "bad task data generated by workflow {}: must be Shareable but got {}".format(
-                                type(self.current_wf.id), type(task_data)
+                                self.current_wf.id, type(task_data)
```
I'm glad that you caught this!

yanchengnv commented on nvflare/private/fed/server/server_runner.py:
```diff
@@ -358,7 +326,7 @@ def process_submission(self, client: Client, task_name: str, task_id: str, resul
 
         if self.status != "started":
-            self.log_info(fl_ctx, "ignored result submission since server runner is {}".format(self.status))
+            self.log_error(fl_ctx, "ignored result submission since server runner's status is {}".format(self.status))
```
INFO should be okay since this is a normal corner condition.

yanchengnv commented on nvflare/private/fed/server/server_runner.py:
```diff
@@ -394,14 +362,14 @@ def process_submission(self, client: Client, task_name: str, task_id: str, resul
             try:
                 if self.current_wf is None:
-                    self.log_info(fl_ctx, "There's no current workflow - dropped submission.")
+                    self.log_error(fl_ctx, "no current workflow - dropped submission.")
```
INFO is okay here.

yanchengnv reviewed (changes requested):
Good job in not only finding/fixing log messages, but also in fixing docstrings and some other bugs.

YuanTingHsieh commented on nvflare/private/fed/server/server_runner.py:
```diff
@@ -201,77 +201,45 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> (str, str,
-            # the client is on a different RUN
-            self.log_info(fl_ctx, "invalid task request: not the same run_number, asked client to end run")
+            # the client is in a different RUN
+            self.log_error(fl_ctx, "invalid task request: not the same run_number - asked client to end the run")
```
changed to INFO.

YuanTingHsieh commented:
Changed the PR as suggested:

- Logs related to "try_again" are changed to DEBUG level
- Logs that related to mismatch run number or context that will lead to "end of run" are changed to INFO level
- Removed unused codes

holgerroth reviewed (dismissed):
LGTM

ZiyueXu77 reviewed (changes requested):
Most messages muted succesfully, two left:

2022-03-08 09:43:44,806 - Communicator - INFO - Received from example_project server  (417 Bytes). getTask time: 0.0034685134887695312 seconds
2022-03-08 09:43:44,808 - FederatedClient - INFO - pull_task completed. Task name:__try_again__ Status:True 

I think we can use 417 Bytes and __try_again__ as criterion to mute these two

ZiyueXu77 reviewed (approved):
Works fine

YuanTingHsieh commented:
Move changes into #289
